### PR TITLE
Added Overlay Widget for saving state

### LINKED ISSUE #311: Add a Saving State
**Tell us about the task you want to perform and are unable to do so because the feature is not available**
Currently when the user clicks on the Save button ... The collection is saved and a notification toast is shown. When the collection is large, this process takes some time & the user has no clue what is happening. So for better UX, a dialog with save animation should be displayed while the saving is in progress.

### PR DESCRIPTION
**Pull request by Tilakraj-B**
## PR Description
Removed the ScaffoldMessenger and added Overlay Widget for displaying the waiting screen while saving for better UX.

## Related Issues

- Related Issue #311 
- Closes #311 

### Checklist
- [X] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [X] I have updated my branch and synced it with project `main` branch before making this PR
- [X] I have run the tests (`flutter test`) and all tests are passing

## Added/updated tests?
added tests for the new components added

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests have not been included_

### CODE REVIEW

**Comment by animator** [comment]
@Tilakraj-B What is the source of `completed.json` & `saving.json`? 
Post the URLs.

**Comment by Tilakraj-B** [comment]
The source is lottiefiles.com
complete.json URL : https://lottiefiles.com/animations/examination-done-cF8EOEIeFo
saving.json URL : https://lottiefiles.com/animations/save-OMZ1ESz3LP

File: lib/widgets/overlay_widget.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+              )
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class CompletedOverlay extends StatelessWidget {
```
Comment (animator) [inline lib/widgets/overlay_widget.dart:62]: No need to two separate `SavingOverlay` & `CompletedOverlay` as they have a lot of duplicate lines. Create a single one which handles both cases and modify the test.

File: lib/screens/home_page/collection_pane.dart [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:apidash/consts.dart';
+import 'package:apidash/models/models.dart';
 import 'package:apidash/providers/providers.dart';
+import 'package:apidash/widgets/overlay_widget.dart';
 import 'package:apidash/widgets/widgets.dart';
-import 'package:apidash/models/models.dart';
-import 'package:apidash/consts.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
```
Comment (animator) [inline lib/screens/home_page/collection_pane.dart:7]: Revert this rearranging of imports. It is not correct.

**Comment by animator** [comment]
Added comments above.

File: lib/consts.dart [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import 'dart:io';
 import 'dart:convert';
+import 'dart:io';
+
+import 'package:davi/davi.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:google_fonts/google_fonts.dart';
-import 'package:davi/davi.dart';
```
Comment (animator) [inline lib/consts.dart:8]: Rearrangement of imports not required.

**Comment by animator** [comment]
One more comment ☝️

**Comment by animator** [comment]
Is the code ready for review?

**Comment by Tilakraj-B** [comment]
yes, made the changes you asked.

**Comment by animator** [comment]
LGTM. Thanks 🚀

**Pull request merged**